assignment1.dart

```diff
@@ -0,0 +1,124 @@
+import 'dart:io';
+
+enum type{
+  raw , manufactured , imported
+}
+
+class Item{      // class name shoud be generic
+  int price;
+  String itemName ;
+  int quantity; 
+  type t1; 
+       // use enum (here)
+
+  Item({required this. price , required this.itemName , required this.quantity , required this.t1}) ;  // can be shortened(redundancy)(done)
+
+  
+ double p(){          /// should i replace this logic by switch- case(?) ==>yes have to use switch case
+ 
+    double currPrice = 0;
+
+    if(this.t1 == type.raw){
+      currPrice = (0.125*this.price)*this.quantity  + this.price*this.quantity;  
+    }
+    else if(this.t1 == type.manufactured){
+       double p1 = (0.125*this.price)*this.quantity ;
+       currPrice = (0.02*p1) + p1 + this.price*this.quantity; 
+    }
+    else{
+
+       double imported = (0.10 * this.price)*this.quantity +this.price*this.quantity;
+
+       if( imported <= 100){
+        imported += 5;
+       }
+       else{if( imported >100 && imported <= 200){
+         imported += 10;
+       }
+       else if(imported > 200){
+         imported = (0.10*imported) + imported;
+       }
+       currPrice = imported;
+
+    }
+
+ }
+ return currPrice;
+
+}
+}
+
+
+
+void main(){
+
+while(true){
```

Comment:
Make a menu-driven program to make it more interactive where user can chose different actions to perform and has access to view the details of previously added items.